game/theater/theatergroundobject.py

```diff
@@ -481,18 +502,77 @@ def __init__(
             dcs_identifier="AA",
             sea_object=False,
         )
-        # Set by the SAM unit generator if the generated group is compatible
-        # with Skynet.
-        self.skynet_capable = False
+        self.aa_ranges: List[AirDefenseRange] = [aa_range]
+
+    # Returns the name of one single group
+    # It also changes the group name to a skynet readable
+    # If the skynet conditions are true (skynet_capability_for_range)
+    def single_group_name(self, group_index: int) -> str:
+
+        if self.is_skynet_capable(group_index):
+            # Return regular name if not skynet_capable
+            # Depends on the skynet_capability_for_range and
+            # if the search radar of the unit is alive
+            return self.skynet_group_name(group_index)
+
+        return super().single_group_name(group_index)
+
+    def set_groups(self, groups: List[Group], ranges: List[AirDefenseRange]):
+        self.groups = groups
+        self.aa_ranges = ranges
+
+    # this function checks if a AA Group should be added to the skynet
+    # based on the AirDefenseRange of the group
+    # The docs recommend to not add AAA at all to skynet
+    # and SHORAD only if its used as Point defence
+    # Otherwise they will not be as efficient as regular
+    def is_skynet_capable(self, group_index: int) -> bool:
+        is_point_defence = group_index > 0
```

Comment:
This seems like a fragile assumption, and I think incorrectly excludes rapiers from skynet.

And in the future we might add separate groups for things like ammo trucks, and those certainly aren't PD :)